flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultSlotStatusSyncer.java

```diff
@@ -308,6 +309,14 @@ private boolean syncAllocatedSlotStatus(SlotStatus slotStatus, TaskManagerInfo t
         }
     }
 
+    @Override
+    public void freeInactiveSlots(JobID jobId, TaskExecutorConnection taskExecutorConnection) {
+        checkStarted();
+        taskExecutorConnection
+                .getTaskExecutorGateway()
+                .freeInactiveSlots(jobId, taskManagerRequestTimeout);
+    }
```

Comment:
Shall we eagerly notify the task manager tracker and resource tracker about the freeing of the slots? Or we rely on task managers to report the slots becoming available later to keep the states consistent? I personally lean towards eagerly sync the status. Either way, it should be documented.